src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -501,28 +393,112 @@ private static MavenArtifactType GetMavenArtifactType(string fileName)
                 // Split the version content into its individual parts.
                 // [0] - The version
                 // [1] - The date it was published
-                // [2] - The time it waas published
-                // [3] - The download count
+                // [2] - The time it was published
+                // [3] - The file size in bytes
                 string[] versionParts = versionContent.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
                 if (DateTime.TryParse($"{versionParts[1]} {versionParts[2]}", out DateTime publishDateTime))
                 {
                     return publishDateTime;
                 }
             }
-            else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+
```

Comment:
Could the above be combined with GetArtifactDownloadUris_DirectoryIndexStrategyAsync somehow?